docs/source/howto/run_codes.rst

```diff
@@ -357,24 +357,215 @@ After :ref:`setting up your computer <how-to:run-codes:computer>` and :ref:`sett
+            'parameters': '2c20fdc49672c3505cebabacfb9b1258e71e7baae5940a80d25837bee0032b59',
+            'structure': 'c0f1c1d1bbcfc7746dcf7d0d675904c62a5b1759d37db77b564948fa5a788769',
+            'parent_calc_folder': 'e375178ceeffcde086546d3ddbce513e0527b5fa99993091b2837201ad96569c'
+        }
+    ]
+
+
+.. _how-to:run-codes:caching:configure:
+
+How to configure caching
+------------------------
+
+Class level
+...........
+
+Besides an on/off switch per profile, the ``.aiida/cache_config.yml`` provides control over caching at the level of specific calculations using their corresponding entry point strings (see the output of ``verdi plugin list aiida.calculations``):
+
+.. code-block:: yaml
+
+    profile-name:
+        default: False
+        enabled:
+            - aiida.calculations:quantumespresso.pw
+        disabled:
+            - aiida.calculations:templatereplacer
+
+In this example, caching is disabled by default, but explicitly enabled for calculaions of the ``PwCalculation`` class, identified by its corresponding ``aiida.calculations:quantumespresso.pw`` entry point string.
+It also shows how to disable caching for particular calculations (which has no effect here due to the profile-wide default).
+
+For calculations which do not have an entry point, you need to specify the fully qualified Python name instead.
+For example, the ``seekpath_structure_analysis`` calcfunction defined in ``aiida_quantumespresso.workflows.functions.seekpath_structure_analysis`` is labelled as ``aiida_quantumespresso.workflows.functions.seekpath_structure_analysis.seekpath_structure_analysis``.
+From an existing :class:`~aiida.orm.nodes.process.calculation.CalculationNode`, you can get the identifier string through the ``process_type`` attribute.
+
+The caching configuration also accepts ``*`` wildcards.
+For example, the following configuration enables caching for all calculation entry points defined by ``aiida-quantumespresso``, and the ``seekpath_structure_analysis`` calcfunction.
+Note that the ``*.seekpath_structure_analysis`` entry needs to be quoted, because it starts with ``*`` which is a special character in YAML.
+
+.. code-block:: yaml
+
+    profile-name:
+        default: False
+        enabled:
+            - aiida.calculations:quantumespresso.*
+            - '*.seekpath_structure_analysis'
+
+You can even override a wildcard with a more specific entry.
+The following configuration enables caching for all ``aiida.calculation`` entry points, except those of ``aiida-quantumespresso``:
+
+.. code-block:: yaml
+
+    profile-name:
+        default: False
+        enabled:
+            - aiida.calculations:*
+        disabled:
+            - aiida.calculations:quantumespresso.*
+
+
+Instance level
+..............
+
+Even when caching is turned off for a given calculation type, you can enable it on a case-by-case basis by using the :class:`~aiida.manage.caching.enable_caching` context manager for testing purposes:
```

Comment:
```suggestion
You can enable caching on a case-by-case basis by using the :class:`~aiida.manage.caching.enable_caching` context manager regardless of the profile settings:
```

Reply:
Shouldn't we here not also rather maintain the style of your preceding comment? I.e., there you changed "You can even override.." into "Any entry with ... is overridden". I actually also prefer that style. Should we change the sentence here rather in something like:
> Caching can be enabled or disabled on a case-by-case basis by using the ....  regardless of the profile settings

Reply:
Good call.